pkg/util/validation/limits.go

```diff
@@ -422,3 +422,18 @@ func (o *Overrides) getOverridesForUser(userID string) *Limits {
 	}
 	return o.defaultLimits
 }
+
+// MaxQueriersPerTenants is returning the smallest maxQueriers value of all
+// given tenant IDs. A value of 0 means unlimted queriers, so the method will
+// return 0 only if all inputs have a limit of 0. Otherwise it is the minimum
+// of > 0 values.
+func MaxQueriersPerTenants(tenantIDs []string, maxQueriersPerTenant func(string) int) int {
```

Comment:
This is one way of doing it, but I believe would be easier to use this was on `o *Overrides` so you don't need to pass `maxQueriersPerTenant` in input.

Reply:
I would suggest different name: `MinimumOfMaxQueriersPerTenants` or `SmallestMaxQueriersPerTenant`. Now it looks like returning "max" value.

Reply:
As discussed on slack, I have removed the method again from override and implement more generic helpers. Not that I am tottally happy with their names, but they should include all previous feedback about naming them:

9826281d5fd8726c5b8db145710cff2d8b9a0e78
0667242d7e0f347a350e8be367140d0f64524d11